### DISCUSSION TITLE
API Calls to dcim/interfaces timing out (502 bad gateway) or JSON parse errors

### DISCUSSION THREAD

**Discussion by mmackay21724** [netbox-community/netbox#8856] [Q&A]
Fairly consistently getting errors on API GET
```/api/dcim/interfaces/?vlan_id=2681```

This is caused by a timeout due to the length taken for the query.  I get most 502 - bad gateway errors but also sometimes we see a JSON parse error which looks to be the result or a worker reset causing the result data to be messed up.

If I do this directly in a browser, in swagger or as an API call in python using pynetbox I get the same result.
Pynet shows the result as <class 'pynetbox.core.response.RecordSet'>
If I debug the result that times out as well.. 

Any suggestions welcome.

Mike

**Comment by jeremystretch** [comment]
1. Use brief mode when retrieving the interfaces (`GET /api/dcim/interfaces/?vlan_id=2681&brief=True`)
2. Increase your WSGI timeout
